File: IND/app/src/main/java/com/example/individual1/MainActivity.java
package com.example.individual1;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.webkit.WebView;
import android.webkit.WebViewFragment;

import com.example.individual1m5.databinding.ActivityMainBinding;
import com.example.individual1m5.databinding.FragmentWebBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        View view = binding.getRoot();
        setContentView(view);

        binding.btir.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                //Se obtiene el texto desde el textfield
                String url = binding.textField.getEditText().getText().toString();
                //Se define webview, se da soporte a fragment manager, se indica ubicacion de fragmento, se muestra y se indica ubicacion del webview
                WebView webView = (WebView)getSupportFragmentManager().findFragmentById(R.id.fragmentContainerView3).getView().findViewById(R.id.wv1);
                //se carga la string url
                webView.loadUrl(url);

            }
        });

        binding.button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                finishAffinity();
            }
        });


    }
}

File: IND (8)/app/src/main/java/com/example/individual8/MainActivity.kt
package com.example.individual8

import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.individual8.databinding.ActivityMainBinding

class MainActivity() : AppCompatActivity() {
    private var binding: ActivityMainBinding? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view: View = binding!!.root
        setContentView(view)
        binding!!.add.setOnClickListener(View.OnClickListener {
            if (!validate()) {
                val resultado = binding!!.et1.text.toString().trim { it <= ' ' }
                    .toInt() + binding!!.et2.text.toString().trim { it <= ' ' }.toInt()
                binding!!.result.text = Integer.toString(resultado)
            } else {
                Toast.makeText(this@MainActivity, "Incorrect Numbers", Toast.LENGTH_LONG).show()
            }
        })
        binding!!.subtract.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View) {
                if (!validate()) {
                    val resultado = binding!!.et1.text.toString().trim { it <= ' ' }
                        .toInt() - binding!!.et2.text.toString().trim { it <= ' ' }.toInt()
                    binding!!.result.text = Integer.toString(resultado)
                } else {
                    Toast.makeText(this@MainActivity, "Incorrect Numbers", Toast.LENGTH_LONG).show()
                }
            }
        })
        binding!!.divide.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View) {
                if (validate()) {
                    Toast.makeText(this@MainActivity, "Incorrect Numbers", Toast.LENGTH_LONG).show()
                } else if (binding!!.et2.text.toString().trim { it <= ' ' }.toInt() == 0) {
                    Toast.makeText(
                        this@MainActivity,
                        "It is impossible to divide by 0",
                        Toast.LENGTH_LONG
                    ).show()
                } else {
                    val resultado = binding!!.et1.text.toString().trim { it <= ' ' }
                        .toInt() / binding!!.et2.text.toString().trim { it <= ' ' }.toInt()
                    binding!!.result.text = Integer.toString(resultado)
                }
            }
        })
        binding!!.multiply.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View) {
                if (!validate()) {
                    val resultado = binding!!.et1.text.toString().trim { it <= ' ' }
                        .toInt() * binding!!.et2.text.toString().trim { it <= ' ' }.toInt()
                    binding!!.result.text = Integer.toString(resultado)
                } else {
                    Toast.makeText(this@MainActivity, "Incorrect Numbers", Toast.LENGTH_LONG).show()
                }
            }
        })
        binding!!.exit.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View) {
                finish()
            }
        })
    }

    fun validate(): Boolean {
        var error = false
        if (binding!!.et1.text.toString().trim { it <= ' ' }
                .isEmpty() || binding!!.et2.text.toString().trim { it <= ' ' }
                .isEmpty()) {
            error = true
        }
        return error
    }
}